### ISSUE TITLE
getSubclasses is empty after fromJson

### ISSUE THREAD

**Issue by rendaw** [classgraph/classgraph#314]
Here's a minimal example:
```
	public static void main(String[] args) {
		ScanResult result = new ClassGraph().verbose().enableClassInfo().whitelistPackages("test1").scan();
		System.out.format("found: %s\n", result.getSubclasses(A.class.getName()));
		result = ScanResult.fromJSON(result.toJSON());
		System.out.format("found 2: %s\n", result.getSubclasses(A.class.getName()));
	}
```
A is an empty class, B extends A is also empty.

This prints:
```
$ (cd test1/target && /usr/lib/jvm/java-11-openjdk/bin/java -p dependency -m test1/test1.Main)
Feb 11, 2019 4:30:59 AM nonapi.io.github.classgraph.utils.LogNode flush
INFO: 2019-02-11T04:30:59.387+0900	ClassGraph	ClassGraph version 4.6.33-SNAPSHOT
2019-02-11T04:30:59.399+0900	ClassGraph	Operating system: Linux 4.19.2-arch1-1-ARCH amd64
2019-02-11T04:30:59.407+0900	ClassGraph	Java version: 11.0.1 / 11.0.1+13 (Oracle Corporation)
2019-02-11T04:30:59.407+0900	ClassGraph	Java home: /usr/lib/jvm/java-11-openjdk
2019-02-11T04:30:59.417+0900	ClassGraph	ScanSpec:
2019-02-11T04:30:59.424+0900	ClassGraph	-- packageWhiteBlackList: whitelist: [test1]
2019-02-11T04:30:59.424+0900	ClassGraph	-- packagePrefixWhiteBlackList: whitelistPrefixes: [test1.]
2019-02-11T04:30:59.424+0900	ClassGraph	-- pathWhiteBlackList: whitelist: [test1/]
2019-02-11T04:30:59.424+0900	ClassGraph	-- pathPrefixWhiteBlackList: whitelistPrefixes: [test1/]
2019-02-11T04:30:59.424+0900	ClassGraph	-- classWhiteBlackList: 
2019-02-11T04:30:59.424+0900	ClassGraph	-- classfilePathWhiteBlackList: 
2019-02-11T04:30:59.424+0900	ClassGraph	-- classPackageWhiteBlackList: 
2019-02-11T04:30:59.424+0900	ClassGraph	-- classPackagePathWhiteBlackList: 
2019-02-11T04:30:59.425+0900	ClassGraph	-- moduleWhiteBlackList: 
2019-02-11T04:30:59.425+0900	ClassGraph	-- jarWhiteBlackList: 
2019-02-11T04:30:59.425+0900	ClassGraph	-- classpathElementResourcePathWhiteBlackList: 
2019-02-11T04:30:59.425+0900	ClassGraph	-- libOrExtJarWhiteBlackList: 
2019-02-11T04:30:59.425+0900	ClassGraph	-- performScan: true
2019-02-11T04:30:59.425+0900	ClassGraph	-- scanJars: true
2019-02-11T04:30:59.425+0900	ClassGraph	-- scanNestedJars: true
2019-02-11T04:30:59.425+0900	ClassGraph	-- scanDirs: true
2019-02-11T04:30:59.425+0900	ClassGraph	-- scanModules: true
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableClassInfo: true
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableFieldInfo: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableMethodInfo: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableAnnotationInfo: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableStaticFinalFieldConstantInitializerValues: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableInterClassDependencies: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableExternalClasses: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- enableSystemJarsAndModules: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- ignoreClassVisibility: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- ignoreFieldVisibility: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- ignoreMethodVisibility: false
2019-02-11T04:30:59.426+0900	ClassGraph	-- disableRuntimeInvisibleAnnotations: false
2019-02-11T04:30:59.427+0900	ClassGraph	-- extendScanningUpwardsToExternalClasses: true
2019-02-11T04:30:59.427+0900	ClassGraph	-- enableRemoteJarScanning: false
2019-02-11T04:30:59.427+0900	ClassGraph	-- addedClassLoaders: null
2019-02-11T04:30:59.427+0900	ClassGraph	-- overrideClassLoaders: null
2019-02-11T04:30:59.427+0900	ClassGraph	-- addedModuleLayers: null
2019-02-11T04:30:59.427+0900	ClassGraph	-- overrideModuleLayers: null
2019-02-11T04:30:59.427+0900	ClassGraph	-- overrideClasspath: null
2019-02-11T04:30:59.427+0900	ClassGraph	-- classpathElementFilters: null
2019-02-11T04:30:59.427+0900	ClassGraph	-- initializeLoadedClasses: false
2019-02-11T04:30:59.428+0900	ClassGraph	-- removeTemporaryFilesAfterScan: false
2019-02-11T04:30:59.428+0900	ClassGraph	-- ignoreParentClassLoaders: false
2019-02-11T04:30:59.428+0900	ClassGraph	-- ignoreParentModuleLayers: false
2019-02-11T04:30:59.429+0900	ClassGraph	-- modulePathInfo: --module-path=dependency
2019-02-11T04:30:59.432+0900	ClassGraph	Finding classpath
2019-02-11T04:30:59.432+0900	ClassGraph	-- Finding classpath and modules
2019-02-11T04:30:59.433+0900	ClassGraph	---- System jars:
2019-02-11T04:30:59.440+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.fxml.jar
2019-02-11T04:30:59.440+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/jrt-fs.jar
2019-02-11T04:30:59.440+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.swing.jar
2019-02-11T04:30:59.440+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx-swt.jar
2019-02-11T04:30:59.440+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.controls.jar
2019-02-11T04:30:59.441+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.base.jar
2019-02-11T04:30:59.441+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.media.jar
2019-02-11T04:30:59.441+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.graphics.jar
2019-02-11T04:30:59.441+0900	ClassGraph	------ Scanning disabled for lib or ext jar: /usr/lib/jvm/java-11-openjdk/lib/javafx.web.jar
2019-02-11T04:30:59.473+0900	ClassGraph	---- Found ClassLoaders:
2019-02-11T04:30:59.477+0900	ClassGraph	------ jdk.internal.loader.ClassLoaders$AppClassLoader
2019-02-11T04:30:59.477+0900	ClassGraph	---- Found system modules:
2019-02-11T04:30:59.477+0900	ClassGraph	------ [module java.base, location=jrt:/java.base]
2019-02-11T04:30:59.477+0900	ClassGraph	------ [module java.compiler, location=jrt:/java.compiler]
2019-02-11T04:30:59.477+0900	ClassGraph	------ [module java.datatransfer, location=jrt:/java.datatransfer]
2019-02-11T04:30:59.477+0900	ClassGraph	------ [module java.desktop, location=jrt:/java.desktop]
2019-02-11T04:30:59.477+0900	ClassGraph	------ [module java.logging, location=jrt:/java.logging]
2019-02-11T04:30:59.477+0900	ClassGraph	------ [module java.management, location=jrt:/java.management]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.management.rmi, location=jrt:/java.management.rmi]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.naming, location=jrt:/java.naming]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.prefs, location=jrt:/java.prefs]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.rmi, location=jrt:/java.rmi]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.security.jgss, location=jrt:/java.security.jgss]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.security.sasl, location=jrt:/java.security.sasl]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.smartcardio, location=jrt:/java.smartcardio]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.xml, location=jrt:/java.xml]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module java.xml.crypto, location=jrt:/java.xml.crypto]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.charsets, location=jrt:/jdk.charsets]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.compiler, location=jrt:/jdk.compiler]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.crypto.cryptoki, location=jrt:/jdk.crypto.cryptoki]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.crypto.ec, location=jrt:/jdk.crypto.ec]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.internal.opt, location=jrt:/jdk.internal.opt]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.jartool, location=jrt:/jdk.jartool]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.javadoc, location=jrt:/jdk.javadoc]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.jdeps, location=jrt:/jdk.jdeps]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.jfr, location=jrt:/jdk.jfr]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.jlink, location=jrt:/jdk.jlink]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.localedata, location=jrt:/jdk.localedata]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.management, location=jrt:/jdk.management]
2019-02-11T04:30:59.478+0900	ClassGraph	------ [module jdk.management.jfr, location=jrt:/jdk.management.jfr]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.naming.dns, location=jrt:/jdk.naming.dns]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.naming.rmi, location=jrt:/jdk.naming.rmi]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.security.auth, location=jrt:/jdk.security.auth]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.security.jgss, location=jrt:/jdk.security.jgss]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.unsupported, location=jrt:/jdk.unsupported]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.unsupported.desktop, location=jrt:/jdk.unsupported.desktop]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module jdk.zipfs, location=jrt:/jdk.zipfs]
2019-02-11T04:30:59.479+0900	ClassGraph	---- Found non-system modules:
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module io.github.classgraph, location=file:///me/test1/target/dependency/classgraph-4.6.33-SNAPSHOT.jar]
2019-02-11T04:30:59.479+0900	ClassGraph	------ [module test1, location=file:///me/test1/target/dependency/test1-1.0.0.jar]
2019-02-11T04:30:59.480+0900	ClassGraph	---- ClassLoaderHandlers:
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.AntClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.EquinoxClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.EquinoxContextFinderClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.FelixClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.JBossClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.WeblogicClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.WebsphereLibertyClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.WebsphereTraditionalClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.OSGiDefaultClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.SpringBootRestartClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.TomcatWebappClassLoaderBaseHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.ParentLastDelegationOrderTestClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.JPMSClassLoaderHandler
2019-02-11T04:30:59.484+0900	ClassGraph	------ nonapi.io.github.classgraph.classloaderhandler.URLClassLoaderHandler
2019-02-11T04:30:59.491+0900	ClassGraph	---- ClassLoader jdk.internal.loader.ClassLoaders$AppClassLoader will be handled by nonapi.io.github.classgraph.classloaderhandler.JPMSClassLoaderHandler@7ff2a664
2019-02-11T04:30:59.491+0900	ClassGraph	---- jdk.internal.loader.ClassLoaders$AppClassLoader delegates to parent jdk.internal.loader.ClassLoaders$PlatformClassLoader with order PARENT_FIRST
2019-02-11T04:30:59.492+0900	ClassGraph	---- ClassLoader jdk.internal.loader.ClassLoaders$PlatformClassLoader will be handled by nonapi.io.github.classgraph.classloaderhandler.JPMSClassLoaderHandler@4d339552
2019-02-11T04:30:59.492+0900	ClassGraph	---- Finding classpath elements in ClassLoaders
2019-02-11T04:30:59.492+0900	ClassGraph	------ Finding classpath elements in ClassLoader jdk.internal.loader.ClassLoaders$PlatformClassLoader
2019-02-11T04:30:59.492+0900	ClassGraph	------ Finding classpath elements in ClassLoader jdk.internal.loader.ClassLoaders$AppClassLoader
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.base
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.compiler
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.datatransfer
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.desktop
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.logging
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.management
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.management.rmi
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.naming
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.prefs
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.rmi
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.security.jgss
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.security.sasl
2019-02-11T04:30:59.493+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.smartcardio
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.xml
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: java.xml.crypto
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.charsets
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.compiler
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.crypto.cryptoki
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.crypto.ec
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.internal.opt
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.jartool
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.javadoc
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.jdeps
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.jfr
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.jlink
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.localedata
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.management
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.management.jfr
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.naming.dns
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.naming.rmi
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.security.auth
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.security.jgss
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.unsupported
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.unsupported.desktop
2019-02-11T04:30:59.494+0900	ClassGraph	-- Skipping non-whitelisted or blacklisted system module: jdk.zipfs
2019-02-11T04:30:59.499+0900	ClassGraph	Finding classpath entries
2019-02-11T04:30:59.500+0900	ClassGraph	-- Opening classpath elements (took 0.000359 sec)
2019-02-11T04:30:59.500+0900	ClassGraph	-- Final classpath element order:
2019-02-11T04:30:59.501+0900	ClassGraph	---- [module io.github.classgraph, location=file:///me/test1/target/dependency/classgraph-4.6.33-SNAPSHOT.jar]
2019-02-11T04:30:59.501+0900	ClassGraph	---- [module test1, location=file:///me/test1/target/dependency/test1-1.0.0.jar]
2019-02-11T04:30:59.501+0900	ClassGraph	Scanning filenames within classpath elements (took 0.019509 sec)
2019-02-11T04:30:59.502+0900	ClassGraph	-- Scanning module io.github.classgraph (took 0.016939 sec)
2019-02-11T04:30:59.518+0900	ClassGraph	---- Found whitelisted classfile: module-info.class
2019-02-11T04:30:59.502+0900	ClassGraph	-- Scanning module test1 (took 0.016996 sec)
2019-02-11T04:30:59.518+0900	ClassGraph	---- Found whitelisted classfile: module-info.class
2019-02-11T04:30:59.519+0900	ClassGraph	---- Found classfile within whitelisted package: test1/A.class
2019-02-11T04:30:59.519+0900	ClassGraph	---- Found classfile within whitelisted package: test1/B.class
2019-02-11T04:30:59.519+0900	ClassGraph	---- Found classfile within whitelisted package: test1/FindMe1.class
2019-02-11T04:30:59.519+0900	ClassGraph	---- Found classfile within whitelisted package: test1/Main.class
2019-02-11T04:30:59.520+0900	ClassGraph	Masking classfiles (took 0.000472 sec)
2019-02-11T04:30:59.522+0900	ClassGraph	Scanning classfiles (took 0.008784 sec)
2019-02-11T04:30:59.525+0900	ClassGraph	-- Parsing classfile jar:file:///me/test1/target/dependency/classgraph-4.6.33-SNAPSHOT.jar!/module-info.class (took 0.005159 sec)
2019-02-11T04:30:59.530+0900	ClassGraph	---- Found class module-info
2019-02-11T04:30:59.525+0900	ClassGraph	-- Parsing classfile jar:file:///me/test1/target/dependency/test1-1.0.0.jar!/module-info.class (took 0.005159 sec)
2019-02-11T04:30:59.530+0900	ClassGraph	---- Found class module-info
2019-02-11T04:30:59.525+0900	ClassGraph	-- Parsing classfile jar:file:///me/test1/target/dependency/test1-1.0.0.jar!/test1/A.class (took 0.005462 sec)
2019-02-11T04:30:59.530+0900	ClassGraph	---- Found class test1.A
2019-02-11T04:30:59.530+0900	ClassGraph	------ Superclass: java.lang.Object
2019-02-11T04:30:59.525+0900	ClassGraph	-- Parsing classfile jar:file:///me/test1/target/dependency/test1-1.0.0.jar!/test1/B.class (took 0.005491 sec)
2019-02-11T04:30:59.530+0900	ClassGraph	---- Found class test1.B
2019-02-11T04:30:59.530+0900	ClassGraph	------ Superclass: test1.A
2019-02-11T04:30:59.525+0900	ClassGraph	-- Parsing classfile jar:file:///me/test1/target/dependency/test1-1.0.0.jar!/test1/FindMe1.class (took 0.005466 sec)
2019-02-11T04:30:59.530+0900	ClassGraph	---- Found class test1.FindMe1
2019-02-11T04:30:59.530+0900	ClassGraph	------ Superclass: java.lang.Object
2019-02-11T04:30:59.525+0900	ClassGraph	-- Parsing classfile jar:file:///me/test1/target/dependency/test1-1.0.0.jar!/test1/Main.class (took 0.005466 sec)
2019-02-11T04:30:59.530+0900	ClassGraph	---- Found class test1.Main
2019-02-11T04:30:59.530+0900	ClassGraph	------ Superclass: java.lang.Object
2019-02-11T04:30:59.531+0900	ClassGraph	Linking related classfiles (took 0.003506 sec)
2019-02-11T04:30:59.546+0900	ClassGraph	Total time: 0.043000000000000003 sec

found: [public class test1.B extends test1.A]
found 2: []
```

**Comment by lukehutch** [comment]
@rendaw I think you're the only person who has ever tried to use the JSON serializer and deserializer :-D  

I have to head out, but I'll take a look at this later today or tomorrow.

**Comment by rendaw** [comment]
Haha no worries, and thanks very much.  I hit the jlink scan issue, so then I thought about doing build-time scanning before jlink and hit this loading the results.

**Comment by lukehutch** [comment]
@rendaw Hopefully you don't hit too many more of these bugs -- I tested the serialization/deserialization code extensively at one point by passing all the unit tests through a serialize/deserialize step, and the tests all passed -- but the code has changed a bit since then. 

However, I should also point out that after writing all the serializer/deserializer code, I realized that for most usecases, serializing the whole `ScanResult` was overkill. It's better to query the `ScanResult` in the build-time code, and write out just the metadata you need into your own format. It's a lot more lightweight to do so, and faster at runtime than deserialization, unless the JSON doc for the `ScanResult` is small.

**Comment by lukehutch** [comment]
@rendaw Fixed in master. Thanks for the report!

**Comment by lukehutch** [comment]
Released in 4.8.0.

https://github.com/classgraph/classgraph/releases/tag/classgraph-4.8.0

Thanks for the bug report!
